otherlibs/dune-rpc/private/dune_rpc_private.mli

```diff
@@ -121,11 +121,35 @@ module Loc : sig
     }
 end
 
-module Error : sig
+module Diagnostic : sig
+  type severity =
+    | Error
+    | Warning
+
   type t =
     { target : string option
     ; message : string
     ; loc : Loc.t option
+    ; severity : severity option
+    }
+end
+
+module Build : sig
```

Comment:
@tmattio have a look and see if this signature covers everything we need. I'll then follow up with an implementation of these rpc calls.

Reply:
I don't have a good idea of the requirements at the moment, so I would suggest that we start by listing the use cases in `vscode`/`lsp` that we want to implement using Dune RPC, and derive the spec of requests from them. I can start working on that next week.

Without these, one requirement that I see is to have the list of output files/artifacts for the build. This could be useful for the work on PPX analysis and will help in implementing build/debug commands. Not sure if this signature is the right place for it though. Maybe this could also be used to fix https://github.com/ocamllabs/vscode-ocaml-platform/issues/524

Reply:
For now, I think it's enough to focus on the watch mode integration. The other features are much less important.

> Without these, one requirement that I see is to have the list of output files/artifacts for the build. This could be useful for the work on PPX analysis and will help in implementing build/debug commands. Not sure if this signature is the right place for it though.

It's the right place to do it.  For ppx, I suppose we could add a declaration like:

```ocaml
val pped : (string, string) request (* module path -> preprocessed output *)
```

To give the preprocessed output. It's something that will need to be delayed until watch mode is done though.

> Maybe this could also be used to fix ocamllabs/vscode-ocaml-platform#524

Let's see what @voodoos thinks about this issue first